tests/arch/x86/arch_interface/src/main.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <sys/arch_interface.h>
+
+#define EFLAGS_IF_MASK 0X200
+/* Get the bit value of IRQ status */
+#define CHECK_IF_BIT(x, y) {        \
+	x = irq_lock();                 \
+	y = (x & EFLAGS_IF_MASK);       \
+	irq_unlock(x);                  \
+}
+static struct k_timer idle_timer;
+static void idle_timer_expiry_function(struct k_timer *timer_id)
+{
+	/* nothing to do */
+}
+
+/**
+ * @brief Verify arch interface of timing functions
+ *
+ * @details Initializing arch timing and start it,
+ * then catch the freq value, it is successful if
+ * the return value is non-zero, and verify
+ * arch_timing_cycles_to_ns_avg API. Then stop it.
+ *
+ * @see arch_timing_freq_get_mhz()
+ * arch_timing_cycles_to_ns_avg()
+ */
+void test_arch_timing_func(void)
+{
+	uint32_t freq = 0;
+	uint64_t cycle_to_ns = 0;
+
+	arch_timing_init();
+	arch_timing_start();
+
+	freq = arch_timing_freq_get_mhz();
+	cycle_to_ns = arch_timing_cycles_to_ns_avg(1000, 10);
+	arch_timing_stop();
+
+	zassert_true(freq != 0, "Failed to call arch_timing_freq_get_mhz().");
+	zassert_true(cycle_to_ns != 0, "failed to call arch_timing_cycles_to_ns_avg");
+}
+
+/**
+ * @brief Verify the bit value of IF in EFLAGS register is not changed
+ * before and after call CPU idle function.
+ *
+ * @details Get the value of EFLAGES, and then call k_cpu_atomic_idle
+ * to make CPU idle, and wake up the CPU by timer interrupt, then compare
+ * the value of IF before and after call CPU idle function. Locked
+ * the IRQ and repeat that.
+ *
+ * @see k_cpu_atomic_idle()
+ */
+void test_kernel_cpu_idle_no_impact_if(void)
```

Comment:
What functionalities are you trying to test here? Why do you need to test if eflags changes?????

Reply:
I want to test that the function k_cpu_atomic_idle can record the state of the IF bit through the key, and restore the level of the IF bit to the state before the call k_cpu_atomic_idle whatever the IF bit is 1 or 0. Do you think this test case is making sense? Thanks.

Reply:
No. Because if this function does not work correctly, the atomic idle test in `tests/kernel/context` would fail or even hang.

Reply:
I know the test cases in `tests/kernel/context`, but it still does not cover the condition that IF bit is 0 before calling the API k_cpu_atomic_idle, it means the command `cli` is still not tested, I add this test case is to test this branch.

Reply:
So this is done only for the sake of code coverage and nothing to do with it's functionality?

Reply:
Yes, this is module testing, and the test case is designed by the source code logic, so no functionality here. And I will also consolidate this into `tests/kernel/context`?